docs/topics/downloader-middleware.rst

```diff
@@ -922,13 +922,118 @@ RobotsTxtMiddleware
     To make sure Scrapy respects robots.txt make sure the middleware is enabled
     and the :setting:`ROBOTSTXT_OBEY` setting is enabled.
 
+    This middleware has to be combined with a robots.txt_ parser.
+
+    Scrapy ships with support for the following robots.txt_ parsers:
+
+    * :ref:`RobotFileParser <python-robotfileparser>` (default)
+    * :ref:`Reppy <reppy-parser>`
+    * :ref:`Robotexclusionrulesparser <rerp-parser>`
+
+    You can change the robots.txt_ parser with the :setting:`ROBOTSTXT_PARSER`
+    setting. Or you can also :ref:`implement support for a new parser <support-for-new-robots-parser>`.
+
 .. reqmeta:: dont_obey_robotstxt
 
 If :attr:`Request.meta <scrapy.http.Request.meta>` has
 ``dont_obey_robotstxt`` key set to True
 the request will be ignored by this middleware even if
 :setting:`ROBOTSTXT_OBEY` is enabled.
 
+.. _python-robotfileparser:
+
+RobotFileParser
+~~~~~~~~~~~~~~~
+
+`RobotFileParser <https://docs.python.org/3.7/library/urllib.robotparser.html>`_ is 
+Python's inbuilt ``robots.txt`` parser. The parser is fully compliant with `Martijn Koster's 
+1996 draft specification <http://www.robotstxt.org/norobots-rfc.txt>`_, with support for
+``Crawl-delay`` and ``Request-rate`` directives also available from Python 3.6. It lacks
+support for wildcard matching. Scrapy uses this parser by default.
+
+In order to use this parser, set:
+
+* :setting:`ROBOTSTXT_PARSER` to ``scrapy.extensions.robotstxtparser.PythonRobotParser``
+
+.. _rerp-parser:
+
+Robotexclusionrulesparser
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Robotexclusionrulesparser <http://nikitathespider.com/python/rerp/>`_ is fully compliant
+with `Martijn Koster's 1996 draft specification <http://www.robotstxt.org/norobots-rfc.txt>`_,
+with support for wildcard matching, ``Crawl-delay`` directive, and ``Sitemap`` directive.
+
+In order to use this parser, set:
+
+* :setting:`ROBOTSTXT_PARSER` to ``scrapy.extensions.robotstxtparser.RerpRobotParser``
+
+.. _reppy-parser:
+
+Reppy parser
+~~~~~~~~~~~~
+
+`Reppy <https://github.com/seomoz/reppy/>`_ is a Python wrapper around `Robots Exclusion
+Protocol Parser for C++ <https://github.com/seomoz/rep-cpp>`_. The parser is fully compliant
+with `Martijn Koster's 1996 draft specification <http://www.robotstxt.org/norobots-rfc.txt>`_,
+with support for wildcard matching, ``Crawl-delay`` directive, and ``Sitemap`` directive.
+Unlike `RobotFileParser <https://docs.python.org/3.7/library/urllib.robotparser.html>`_ and 
+`Robotexclusionrulesparser <http://nikitathespider.com/python/rerp/>`_, it uses the length based 
+rule, in particular for ``Allow`` and ``Disallow`` directives, where the most specific rule based 
+on the length of the path trumps the less specific (shorter) rule.
+
+In order to use this parser, set:
+
+* :setting:`ROBOTSTXT_PARSER` to ``scrapy.extensions.robotstxtparser.ReppyRobotParser``
```

Comment:
I'm not sure scrapy.extensions is a right place for parsers, because extensions have a specific meaning - they are put into EXTENSIONS setting, created by Scrapy when spider starts, etc.; they are also expected to subscribe to signals. Here these backends are created by the middleware, similar to storages for FilePipeline, or exporters for feed exports. 

Not sure about the best place for this. `scrapy.robotstxt`?  @dangra @Gallaecio what do you think?

Reply:
I moved all the parser interface classes to `scrapy.robotstxt`.